klasses/utils.py

```diff
@@ -0,0 +1,53 @@
+"""Utility functions for Klasses"""
+
+from klasses.models import BootcampRunCertificate, BootcampRunEnrollment
+
+
+def generate_single_certificate(user, bootcamp_run):
+    """Generates certificate for a single user if its enrollment is active"""
+    if (
+        user
+        and bootcamp_run
+        and BootcampRunEnrollment.objects.filter(
+            user=user, bootcamp_run=bootcamp_run, active=True
+        ).exists()
+    ):
+        instance, _ = BootcampRunCertificate.all_objects.update_or_create(
+            user=user, bootcamp_run=bootcamp_run
+        )
+        return instance
+
+
+def generate_batch_certificates(bootcamp_run):
+    """Generates certificates for all the users who have active enrollments in a bootcamp-run"""
+    if bootcamp_run:
+        return BootcampRunCertificate.objects.bulk_create(
+            [
+                BootcampRunCertificate(user=enrollment.user, bootcamp_run=bootcamp_run)
+                for enrollment in BootcampRunEnrollment.objects.filter(
+                    bootcamp_run=bootcamp_run, active=True
+                )
+                if not BootcampRunCertificate.all_objects.filter(
+                    user=enrollment.user, bootcamp_run=bootcamp_run
+                ).exists()
+            ],
+            ignore_conflicts=True,
+        )
+
+
+def revoke_certificate(user, bootcamp_run):
+    """Revokes certificate for the given user and bootcamp-run"""
+    if user and bootcamp_run:
+        certificate = BootcampRunCertificate.all_objects.filter(
+            user=user, bootcamp_run=bootcamp_run
+        ).first()
+        return certificate.revoke() if certificate else None
```

Comment:
What happens if no certificate is found?

Reply:
I have added a lot of logging now, hope it will help. You may have a look.